pkg/pubsub/invoker/invoker.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2019 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package invoker
+
+import (
+	"errors"
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/transport"
+	cepubsub "github.com/cloudevents/sdk-go/pkg/cloudevents/transport/pubsub"
+	"github.com/google/uuid"
+	"golang.org/x/net/context"
+
+	"github.com/GoogleCloudPlatform/cloud-run-events/pkg/apis/pubsub/v1alpha1"
+	"github.com/GoogleCloudPlatform/cloud-run-events/pkg/kncloudevents"
+)
+
+// TODO: not ready.
+
+// Invoker implements the Pub/Sub invoker to deliver Pub/Sub messages from a
+// pre-existing topic with a set of pre-existing subscriptions to the set of
+// Channel Subscribers.
+type Invoker struct {
+	// ProjectID is the pre-existing eventing project id to use.
+	ProjectID string
+
+	// inbound is the cloudevents client to use to receive events.
+	inbound          cloudevents.Client
+	inboundTransport *cepubsub.Transport
```

Comment:
Where is inboundTransport set?

Reply:
it is removed in a followup. It was being used to delete subscriptions but I decided it was a bad idea.